pyqtgraph/opengl/items/GLTextItem.py

```diff
@@ -0,0 +1,74 @@
+from typing import List, Union
+from OpenGL.GL import *
+from OpenGL.GLUT import *
+import numpy as np
+from pyqtgraph.opengl.GLGraphicsItem import GLGraphicsItem
+
+class GLTextItem(GLGraphicsItem):
+    """Draws text in 3D."""
+
+    pos:Union[np.ndarray, List[float]] = np.array([0.0, 0.0, 0.0])
+    color:np.ndarray = np.array([1.0, 1.0, 1.0, 1.0])
+    text:str = ''
+    font = GLUT_BITMAP_HELVETICA_18
+
+    def __init__(self, **kwds):
+        """All keyword arguments are passed to setData()"""
+        GLGraphicsItem.__init__(self)
+        glopts = kwds.pop('glOptions', 'additive')
+        self.setGLOptions(glopts)
+        self.setData(**kwds)
+        glutInit()
+    
+    def setData(self, **kwds):
+        """
+        Update the data displayed by this item. All arguments are optional;
+        for example it is allowed to update text while leaving colors unchanged, etc.
+
+        ====================  ==================================================
+        **Arguments:**
+        ------------------------------------------------------------------------
+        pos                   (3,) array of floats specifying text location.
+        color                 (4,) array of floats (0.0-1.0) or
+                              tuple of floats specifying
+                              a single color for the entire item.
+        text                  String to display.
+        font                  GLUT font. (Default: GLUT_BITMAP_HELVETICA_18)
+        ====================  ==================================================
+        """
+        args = ['pos', 'color', 'text', 'font']
+        for k in kwds.keys():
+            if k not in args:
+                raise Exception('Invalid keyword argument: %s (allowed arguments are %s)' % (k, str(args)))
+        for arg in args:
+            if arg in kwds:
+                value = kwds[arg]
+                if arg == 'pos':
+                    if isinstance(value, np.ndarray):
+                        if value.shape != (3,):
+                            raise Exception('"pos.shape" must be (3,).')
+                    elif isinstance(value, tuple) or isinstance(value, list):
```

Comment:
we can combine these in one check

```python
elif isinstance(value, (tuple, list)):
```

Reply:
EDIT, actually you can likely combine this with the if check earlier; since `np.ndarray` supports `len()` 

```python
if arg == 'pos':
    if len(value) != 3:
        raise Exception(....)
```

Reply:
> we can combine these in one check
> 
> ```python
> elif isinstance(value, (tuple, list)):
> ```

Thanks. I have increased my knowledge.
I fixed it in [1ba5203](https://github.com/pyqtgraph/pyqtgraph/pull/1776/commits/1ba52032d743a7ee03941576a40f9893fe13fa04).



> EDIT, actually you can likely combine this with the if check earlier; since `np.ndarray` supports `len()`
> 
> ```python
> if arg == 'pos':
>     if len(value) != 3:
>         raise Exception(....)
> ```

It is true that `np.ndarray` supports `len()`. However, it will pass the check in the following case.
```python
pos = np.array([
    [0.1, 1.2, 5.3, 4.6],
    [4.5, 6.8, 4.7, 8.6],
    [5.6, 8.7, 1.9, 2.0]
])

print(len(pos))
```
```
3
```

Reply:
Ahh that makes sense 👍🏻